Ginger/Ginger/AutomatePageLib/AddActionMenu/ActionsFactory.cs

```diff
@@ -289,10 +309,30 @@ public static void AddActivitiesGroupsFromSRHandler(List<ActivitiesGroup> shared
         {
             foreach (ActivitiesGroup sharedGroup in sharedActivitiesGroupsToAdd)
             {
+                eUserMsgSelection userSelection = eUserMsgSelection.None;
                 ActivitiesGroup droppedGroupIns = (ActivitiesGroup)sharedGroup.CreateInstance(true);
                 businessFlow.AddActivitiesGroup(droppedGroupIns);
                 ObservableList<Activity> activities = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<Activity>();
-                businessFlow.ImportActivitiesGroupActivitiesFromRepository(droppedGroupIns, activities, false);
+                foreach (Activity activityIns in activities)
+                {
```

Comment:
Inside ImportActivitiesGroupActivitiesFromRepository we must be iterating all activities. So better to check TA there and call the method that you will create from there instead of iterating twice